### PR TITLE
Make executable all files in archive, which marked executable

### PR DESCRIPTION
**Pull request by kgevorkyan**
### What's done:
* Create logic of marking files from archive executable
* Fixes: Save attributes during copying/movement files

### CODE REVIEW

**Review by akuleshov7 — APPROVED** [review]

File: save-cloud-common/src/jvmMain/kotlin/org/cqfn/save/utils/FileUtils.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * @param destinationDir destination directory
+ * @throws FileNotFoundException if source doesn't exists
+ */
+fun moveFileWithAttributes(source: File, destinationDir: File) {
+    if (!source.exists()) {
+        throw FileNotFoundException("Source file $source doesn't exist!")
+    }
+
+    Files.copy(source.toPath(), destinationDir.resolve(source.name).toPath(), StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.COPY_ATTRIBUTES)
+    Files.delete(source.toPath())
```
Comment (petertrr) [inline save-cloud-common/src/jvmMain/kotlin/org/cqfn/save/utils/FileUtils.kt:23]: `Files.move`?

Comment (kgevorkyan) [reply save-cloud-common/src/jvmMain/kotlin/org/cqfn/save/utils/FileUtils.kt:23]: Unfortunately, `Files.move` doesn't support `StandardCopyOption.COPY_ATTRIBUTE` option, so I used this workaround method

Comment (petertrr) [reply save-cloud-common/src/jvmMain/kotlin/org/cqfn/save/utils/FileUtils.kt:23]: Hm, that's strange. They even have it described in doc: `An implementation may also attempt to copy other file attributes but is not required to fail if the file attributes cannot be copied.`

**Review by petertrr — APPROVED** [review]

File: save-cloud-common/src/jvmMain/kotlin/org/cqfn/save/utils/FileUtils.kt (hunk 1 above)
Comment (kgevorkyan) [reply save-cloud-common/src/jvmMain/kotlin/org/cqfn/save/utils/FileUtils.kt:23]: Besides that I met exceptions `unsupported option`, here it's also can be seen https://docs.oracle.com/javase/tutorial/essential/io/move.html, that they support only `REPLACE_EXISTING` and `ATOMIC_MOVE` options

**Pull request merged**